indexer/feature_meta.cpp

```diff
@@ -1,72 +1,33 @@
 #include "indexer/feature_meta.hpp"
 
-namespace
-{
-char HexToDec(char ch)
-{
-  if (ch >= '0' && ch <= '9')
-    return ch - '0';
-  if (ch >= 'a')
-    ch -= 'a' - 'A';
-  if (ch >= 'A' && ch <= 'F')
-    return ch - 'A' + 10;
-  return -1;
-}
+#include "std/algorithm.hpp"
 
-string UriDecode(string const & sSrc)
+namespace feature
 {
-  // This code is based on
-  // http://www.codeguru.com/cpp/cpp/string/conversions/article.php/c12759
-  //
-  // Note from RFC1630: "Sequences which start with a percent
-  // sign but are not followed by two hexadecimal characters
-  // (0-9, A-F) are reserved for future extension"
-
-  string result(sSrc.length(), 0);
-  auto itResult = result.begin();
 
-  for (auto it = sSrc.begin(); it != sSrc.end(); ++it)
+string Metadata::GetWikiURL() const
+{
+  string v = this->Get(FMD_WIKIPEDIA);
+  if (!v.empty())
   {
-    if (*it == '%')
+    auto const colon = v.find(':');
+    if (colon != string::npos)
     {
-      if (distance(it, sSrc.end()) > 2)
-      {
-        char dec1 = HexToDec(*(it + 1));
-        char dec2 = HexToDec(*(it + 2));
-        if (-1 != dec1 && -1 != dec2)
-        {
-          *itResult++ = (dec1 << 4) + dec2;
-          it += 2;
-          continue;
-        }
+      // Spaces and % sign should be replaced in urls.
+      replace(v.begin() + colon, v.end(), ' ', '_');
+      string::value_type percent, pos = colon;
+      string const escapedPercent("%25");
+      while ((percent = v.find('%', pos)) != string::npos) {
```

Comment:
Is it ok to convert string::size_type (return type of string::find) to string::value_type (decltype of percent)?